networkit/cpp/distance/test/DistanceGTest.cpp

```diff
@@ -147,6 +152,91 @@ TEST_F(DistanceGTest, testAStar) {
     testMesh(25, 5);
 }
 
+TEST_P(DistanceGTest, testAdamicAdar) {
+    Aux::Random::setSeed(43, false);
+    auto G = generateERGraph(500, 0.03);
+    G.indexEdges();
+
+    AdamicAdarDistance AAD(G);
+    AAD.preprocess();
+    auto rndEdge = GraphTools::randomEdge(G);
+    EXPECT_GT(AAD.distance(rndEdge.first, rndEdge.second), 0.2);
+    auto res = AAD.getEdgeScores();
+    EXPECT_EQ(G.numberOfEdges(), res.size());
+}
+
+TEST_P(DistanceGTest, testAlgebraicDistance) {
+    Aux::Random::setSeed(42, false);
+    auto G = generateERGraph(500, 0.03);
+    G.indexEdges();
+
+    AlgebraicDistance AGD(G, 10UL, 30UL, 0.5, 0UL, true);
+    AGD.preprocess();
+    node source = GraphTools::randomNode(G);
+    node target = GraphTools::randomNode(G);
+    EXPECT_GT(AGD.distance(source, target), 0.2);
+    auto res = AGD.getEdgeScores();
+    for (auto dist : res) {
+        EXPECT_LE(dist, 1);
+    }
+}
+
+TEST_P(DistanceGTest, testEccentricity) {
+    // only undirected and unweighted implemented
+    if (!isDirected() && !isWeighted()) {
```

Comment:
Here we should use `TEST_F` since only one combination of the parameters is supported.